LDAR_Sim/src/file_processing/output_processing/summary_visualizations.py

```diff
@@ -239,13 +239,108 @@ def plot_probit(
         plt.close(combined_plot)
 
 
+def plot_box_whisker(
+    cost_mit_ratio_data: pd.DataFrame,
+    visualization_dir: Path,
+    visualization_name: str,
+    viz_mapper: SummaryVisualizationMapper,
+):
+    box_plot_properties: dict = viz_mapper.get_boxplot_properties(visualization_name)
+
+    x_label: str = box_plot_properties.pop("x_label")
+    y_label: str = box_plot_properties.pop("y_label")
+    # Group by 'program name' and collect 'mitigation_ratio' values into lists
+    grouped_data = (
+        cost_mit_ratio_data.groupby(output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.PROG_NAME)[
+            output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.MITIGATION_RATIO
+        ]
+        .apply(list)
+        .reset_index(name=output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.MITIGATION_RATIO)
+    )
+    labels = grouped_data[output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.PROG_NAME].tolist()
+    data_to_plot = grouped_data[
+        output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.MITIGATION_RATIO
+    ].tolist()
+
+    figure: plt.Figure = plt.figure(figsize=(11, 7))  # noqa 841
+    ax: plt.Axes = plt.gca()
+    ax.boxplot(data_to_plot, labels=labels, **box_plot_properties)
+
+    ax.set_xlabel(x_label)
+    ax.set_ylabel(y_label)
+    plt.tight_layout()
+
+    save_path: Path = visualization_dir / visualization_name
+    plt.savefig(save_path)
+    plt.close(figure)
+
+
+def plot_stack_bar_chart(
+    cost_data: pd.DataFrame,
+    visualization_dir: Path,
+    visualization_name: str,
+    viz_mapper: SummaryVisualizationMapper,
+):
+    bar_chart_properties: dict = viz_mapper.get_bar_chart_properties(visualization_name)
+    y_axis_formatter: ticker.FuncFormatter | None = viz_mapper.get_axis_formatter(
+        visualization_name
+    )
+    # Extract NET_COST and remove it from cost_data
+    net_cost = cost_data.pop(output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.NET_COST)
+
+    x_label: str = bar_chart_properties.pop("x_label")
+    y_label: str = bar_chart_properties.pop("y_label")
+
+    figure: plt.Figure = plt.figure(figsize=(11, 7))  # noqa 841
+    ax: plt.Axes = plt.gca()
+    cost_data.plot(
+        x=output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.PROG_NAME,
+        kind="bar",
+        stacked=True,
+        ax=ax,
+    )
+    # Plot NET_COST as dots on top of the stacked bar chart
+    program_names = cost_data[output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.PROG_NAME]
+    x_positions = range(len(program_names))  # Assuming program names are on the x-axis
+    ax.scatter(
+        x_positions,
+        net_cost,
+        color="black",
+        zorder=5,
+        label=output_file_constants.COST_SUMMARY_COLUMNS_ACCESSORS.NET_COST,
+    )  # zorder=5 to plot dots above bars
+
+    # Add text labels for each dot to display the actual Net Cost value
+    for x_pos, net_cost_value in zip(x_positions, net_cost):
+        ax.text(
+            x_pos,
```

Comment:
Can we add like 2 or 3 to x_pos and net cost val for the label so it's a little further away from the dot